src/lib/utils/settings/abstract_setting.cpp

```diff
@@ -0,0 +1,20 @@
+#include "abstract_setting.hpp"
+
+#include "hyrise.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+AbstractSetting::AbstractSetting(const std::string& init_name) : name(init_name) {}
+
+void AbstractSetting::enroll() {
+  auto name_copy = name;
```

Comment:
Why do we need the name_copy here?

Reply:
Just a relict of my tries to make the setting registration work :D